pkg/executor/timeout/timeoutexecutor.go

```diff
@@ -0,0 +1,41 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package timeout
+
+type timer interface {
+	Expired() bool
+}
+
+type Executor struct {
+	timer timer
+}
+
+func NewExecutor(timer timer) *Executor {
+	return &Executor{timer: timer}
+}
+
+// Exec will execute the given func when the underlying timer has not expired.
+// In case the timer expired, the command is not executed and nil is returned.
+func (t Executor) Exec(command func() error) error {
```

Comment:
I think that in most cases, this will not be needed.
The caller should not care if it was called or not.

The caller needs to decide if a command should be called or not.
Once a decision was made to call it, this executor will either execute the command successfully or not call it at all. The caller does not care.

If the caller cares, we either need to extend this with a `IsTimeout` or with a special error returned. But I do not think we need it now.

Either way, we should wait for @RamLavi feedback on using it in that PR scenario.

Reply:
I agree with @EdDev. Currently we don't need to know if the timeout has passed. If the need for it ever arises, we can extend the API without issues.